ompi/mca/coll/basic/coll_basic_neighbor_alltoall.c

```diff
@@ -67,7 +68,7 @@ mca_coll_basic_neighbor_alltoall_cart(const void *sbuf, int scount, struct ompi_
         if (MPI_PROC_NULL != srank) {
             nreqs++;
             rc = MCA_PML_CALL(irecv(rbuf, rcount, rdtype, srank,
-                                    MCA_COLL_BASE_TAG_ALLTOALL,
+                                    MCA_COLL_BASE_TAG_NONBLOCKING_BASE - 2 * dim,
```

Comment:
This tag range was reserved (as the name indicates) for nonblocking collectives. If we are using it for something else (as we should in this particular case) I would suggest we rename it to something more meaningful.

Reply:
Yeah. We can add another reserved range to be safe. I can add that to this PR.